perf(formatter): optimize formatting of JSX element/fragment with a single child

Dunqing opened the pull request:
Add `FormatSingleChild` for optimizing JSX elements/fragments with a single meaningful child.

When there is only a single child, we do not need to write the child into a temporary buffer and then take it when formatting. Instead, we can directly write the child into the root buffer.

Also, this can avoid calling costly `best_fitting!` formatting in some situations.

leaysgur commented:
### Merge activity

* **Dec 9, 9:02 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 9, 9:02 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Dunqing committed:
perf(formatter): optimize formatting of JSX element/fragment with a single child (#16631)

Add `FormatSingleChild` for optimizing JSX elements/fragments with a single meaningful child.

When there is only a single child, we do not need to write the child into a temporary buffer and then take it when formatting. Instead, we can directly write the child into the root buffer.

Also, this can avoid calling costly `best_fitting!` formatting in some situations.
